docs/DeveloperGuide.md

```diff
@@ -35,4 +348,212 @@
+
+
+- Test case : `set -12.2`
+- Expected : A message stating that the budget set is of a negative value will be returned. Input budget will not
+be stored.
+
+
+- Other invalid `set` commands: eg. `set 3-3`
+- Expected : Similar to previous, an invalid message with the error will be displayed for the user.
+
+#### Check budget
+1. Checking the amount of spending and the intended budget.
+- Prerequisite :  A budget must be set prior to calling `check` and the budget set cannot be of value 0.
+
+
+- Test case : `check` where budget is more than total expenditures in list.
+- Expected : The amount of money away from the set budget will be displayed with other information such as 
+the total spending, budget and borrowed money.
+
+
+- Test case : `check` where budget is less than total expenditures in list.
+- Expected : Similar to previous test case, amount of money exceeded by and other information will be 
+displayed in the message.
+
+#### Find keyword
+1. Finding keywords under the descriptions column in their list of expenditures
+
+- Test case : `find bus`
+- Prerequisite : There are existing expenditures with the description : `bus`
+- Expected : List of items corresponding to the keyword will be displayed. 
+
+
+- Test case : `find taxi`
+- Prerequisite : There are no existing expenditures with the description : `taxi`
+- Expected : Message showing that no matching records are found in the list.
+
+2. View specific date expenditures under the date column
+
+- Test case : `viewdate 2023-02-20`
+- Prerequisite : There are current expenditures dated 20 Feb 2023.
+- Expected : List of all expenditures with the corresponding date value, as well as the total amount spent
+on that specific date
+
+
+- Test case : `viewdate 2023-02-20`
+- Prerequisite : There are no current expenditures dated 20 Feb 2023.
+- Expected : Similar to previous, but there will not be any items shown in the list. The total amount will
+be shown as 0.
+
+
+- Test case : `viewdate 12 Jan 2021`
+- Expected : Invalid message will be shown with the respective error message, in this case being a 
+date time error.
+
+- Other invalid `viewdate` commands: eg. `viewdate`
+- Expected : Similar to previous, an invalid message with the error will be displayed for the user.
+
+3. View specific type of expenditure under the expenditure column
+
+- Test case : `viewtype transport`
+- Prerequisite : There are current expenditures with the `transport` type.
+- Expected : List of all expenditures under transport expenditure, as well as the total amount spent
+  for that type of expenditure
+
+
+- Test case : `viewtype transport`
+- Prerequisite : There are no current expenditures with the `transport` type.
+- Expected : Similar to previous, but there will not be any items shown in the list. The total amount will
+  be shown as 0.
+
+
+- Test case : `viewtype swimming`
+- Expected : Invalid message will be shown with the respective error message, in this case an
```

Comment:
For each of the error message, it would be good to add an example to give the user a clearer idea on what the error message would look like